guide/topics/discordjs/firstCommand.md

```diff
@@ -0,0 +1,163 @@
+
+The other parameter, a function, is the function that will be invoked when the event is fired. Oftentimes, it takes in parameters specific for that type of event. Our `message` event, only returns 1 parameter: a message object which we will be able to utilize in a second.
+
+You can place the event anywhere between the client initialization and the logging in of the client. The best place for it, would be after the `ready` event to keep the code understandable and clean.
+
+## Responding to the message
+If you set up the code correctly, now, when the code is being run, and a message on any server with your bot is sent, a log should appear in our console.
+
+Next up, we will discuss how we can use the `message` object returned from the event to send a response in the channel that the message has been sent.
+
+`message` object has the following properties in it:
+- author - returns the [User](https://discord.js.org/#/docs/main/stable/class/User) that has sent the message.
+- channel - returns the [TextChannel](https://discord.js.org/#/docs/main/stable/class/TextChannel) that the message has been sent in.
+- content - returns the content of the message
+- guild - returns the [Guild](https://discord.js.org/#/docs/main/stable/class/Guild) (server) of where the message was sent in
+- id - returns the ID of the message - this is especially useful when saving a reference to the message
+
+And a few functions, too:
+- delete() - deletes the message
+- pin() - pins the message to the channel
+- react(emoji) - reacts to the message with an emoji object
+
+There are many more properties in the `message` class, however these are the most used ones. If you'd like to get to know each one of them, you can view it here: [Message Class on discord.js.org](https://discord.js.org/#/docs/main/stable/class/Message)
+
+Having these in mind, we can construct the following code:
+```javascript
+client.on("message", (message) => {
+  if(message.author.bot || !message.guild) return;
+  message.channel.send(`Hello, ${message.author.username}!`);
+});
+```
+
+To make sure that the bot will not fall in a loop responding to it's own messages, we have to add an `if` statement check, to check if the author is a bot. If so, `return` the code. The same goes for the second line, which check's for the guild of the message, as we don't want to listen to DM messages. Now, that we are sure that the bot will not fall into a loop and that guild is present on the object, we can move on.  In the first bit of the last line, we are looking at `message` and then accessing the `channel` property of it. That returns the channel that the message was sent in. Channels have a method `.send(content)` that sends a message to them - and it requires a parameter of what the content of that message should be.
```

Comment:
Also, since you're discussing the `.send()` method here, and this is in the context of a "textable" channel, I would link the documentation for [`TextChannel#send()`](https://discord.js.org/#/docs/main/stable/class/TextChannel?scrollTo=send) for example so readers can see various examples of using the method.

Reply:
I don't think that the condition ``!message.guild`` really help not to get into a loop. Because, this condition only checks if the message has been posted on a guild which cannot really cause a loop. Also, if we keep this condition, we would need to say that it would ignore messages sent in the bot's DM.

Reply:
> I don't think that the condition `!message.guild` really help not to get into a loop. Because, this condition only checks if the message has been posted on a guild which cannot really cause a loop. Also, if we keep this condition, we would need to say that it would ignore messages sent in the bot's DM.

That's a fair statement to make. In this case, you could put both conditions in their own `if` statements? Then you can explain the purpose of both conditions separately since in this tutorial, it only really matters if the message author is a bot or not.